microsoft/FluidFramework
# Proof of concept map incremental plus minimal runtime changes

[AB#3250](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3250)

Create a bare minimal proof of concept of DDS level incremental summaries.

This is an example of how SharedMap could be made incremental.

The partition algorithm is very basic and thus would need to be improved to actually provide a better solution than what shared map can currently do.

The blob partition algorithm is currently just based on the order of the keys, so if we have keys 0-9, and key0 changes, then the whole SharedMapIncremental would be summarized as all the blob partitions are likely to have changed. No summary handles would be generated.

The example test has keys 0-9 in which key9 changes and most of the blobs are handles instead of content.

The SharedMapIncremental blob partition sized was reduced specifically for the test from 8 * 1024 to 140 to make it easier to create multiple blobs.

The test scenario:
1. Set 0-9 keys.
2. Summarize creating summary1
3. Set key 9 to a different value
4. Summarize creating summary2
5. Validate that some summary handles were generated instead of the blobs
6. Validate that creating a new container from summary2 should contain the same values as the original container.
